service/history/hsm/executor.go

```diff
@@ -66,6 +67,9 @@ type Environment interface {
 	// When using AccessRead, the accessor must guarantee not to mutate any state, accessor errors will not cause
 	// mutable state unload.
 	Access(ctx context.Context, ref Ref, accessType AccessType, accessor func(*Node) error) error
+	// Load a history event by key and a token generated by a [Node].
+	// This method should be called outside of the Access function to minimize the lock time for a given key.
+	LoadHistoryEvent(ctx context.Context, key definition.WorkflowKey, token []byte) (*historypb.HistoryEvent, error)
```

Comment:
I think it's a little confusing that `LoadHistoryEvent` and `AddHistoryEvent` are part of separate interfaces. I would expect all event CRUD to be part of the `NodeBackend`. Especially because we are using mutable state as our backend and that contains several methods for getting various history events. Could you give a quick reasoning for the split?

Reply:
The reason is in the docstring. This method should ideally be called outside of an `Access` function to avoid holding a workflow lock for too long.